app/main.py

```diff
@@ -2,0 +1,31 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self. hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}, Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden is True:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+
+    @classmethod
+    def bite(cls, animal: Animal) -> None:
+        if animal.hidden is False and not isinstance(animal, Carnivore):
+            animal.health -= 50
+        if animal.health <= 0:
+            super().alive.remove(animal)
```

Comment:
The use of `super().alive.remove(animal)` is incorrect. `super()` is not needed here since `alive` is a class variable of `Animal`. You should use `Animal.alive.remove(animal)` instead.